src/openms/source/FORMAT/PepXMLFile.cpp

```diff
@@ -281,21 +281,24 @@ namespace OpenMS
         AASequence seq = h.getSequence();
         double precursor_neutral_mass = seq.getMonoWeight();
 
-        Int scan_index;
+        int scan_index = count;
+        int scan_nr = 0;
+        
         if (lookup.empty())
         {
           if (it->metaValueExists("RT_index")) // Setting metaValue "RT_index" in XTandemXMLFile in the case of X! Tandem.
           {
-            scan_index = it->getMetaValue("RT_index");
-          }
-          else
-          {
-            scan_index = count;
+            scan_index = it->getMetaValue("RT_index");            
           }
+          scan_nr = scan_index + 1;          
         }
         else
         {
-          scan_index = lookup.findByRT(it->getRT()) + 1;
+          scan_index = lookup.findByRT(it->getRT());
```

Comment:
That is true, that the index mostly starts with 0. But I would not suggest to increment the scan_index, because the intention of the spectrum lookup is to get the original values for scan_index and scan_nr. These values are necessary that Luciphor is able to match the pepXML output with the spectrum.